logback-core/src/main/java/ch/qos/logback/core/ConsoleAppender.java

```diff
@@ -94,6 +96,20 @@ private OutputStream wrapWithJansi(OutputStream targetStream) {
             addInfo("Enabling JANSI AnsiPrintStream for the console.");
             ClassLoader classLoader = Loader.getClassLoaderOfObject(context);
             Class<?> classObj = classLoader.loadClass(AnsiConsole_CLASS_NAME);
+
+            // check for JAnsi 2
+            final Optional<Method> optOutMethod = Arrays.stream(classObj.getMethods())
+                    .filter(m -> m.getName().equals("out"))
```

Comment:
I think this should check wether the `target == ConsoleTarget.SystemOut` to look for either `out` or `err` method.

Reply:
You are right! Will fix this.